Release notes for October 29 (#3974)
Azure/AKS: CHANGELOG.md

# Azure Kubernetes Service Changelog

## Release 2023-10-29

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

> ### Announcements

Reviewer: [Windows containerd v1.7](https://github.com/Azure/AKS/issues/3975) will be the default container runtime for k8s v1.28+ on AKS Windows nodes. Windows Host Process (HPC) containers is GA in Windows containerd v1.7 and it has some [breaking changes](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support#container-mounts).

* Kubernetes 1.25 is being deprecated at the end of January 2024 and support will transition to our [platform support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#platform-support-policy).
* No new clusters can be created with [Azure AD Integration (legacy)](https://learn.microsoft.com/azure/aks/azure-ad-integration-cli). Existing AKS clusters with Azure Active Directory integration will keep working. All Azure AD Integration (legacy) AKS clusters will be migrated to [AKS-managed Azure AD](https://learn.microsoft.com/azure/aks/managed-azure-ad) automatically starting from December 1st, 2023. We recommend updating your cluster with AKS-managed Azure AD before December 1st, 2023. This way you can manage the API server downtime during non-business hours.

> * Starting January 2024, due to Gatekeeper Upstream removing validation for constraint template contents at create/update time, [the Azure Policy Add-On](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fgovernance%2Fpolicy%2Fconcepts%2Fpolicy-for-kubernetes%23install-azure-policy-add-on-for-aks&data=05%7C01%7Cnehakulkarni%40microsoft.com%7C132aa96f5b9a4108065908dbd718d258%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638340273857129567%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=73Ow8x1Ox8bTw5yOP5P4NIk1R%2FYnaDzhExBBuDbz62g%3D&reserved=0) will now no longer support this. The Azure Policy Add-On will report [‘InvalidConstraint/Template’ compliance reason code](https://learn.microsoft.com/en-us/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons) for detected errors after constraint template admission. This change does not impact [other compliance reason codes](https://learn.microsoft.com/en-us/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons). Customers are encouraged to continue to follow best practices when updating Azure Policy for Kubernetes definitions (i.e. [Gator CLI](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fopen-policy-agent.github.io%2Fgatekeeper%2Fwebsite%2Fdocs%2Fgator%2F&data=05%7C01%7Cnehakulkarni%40microsoft.com%7C132aa96f5b9a4108065908dbd718d258%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638340273857147647%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6VrGYNGtikDsPw3odmyR9WatG4Pe1tlsvDgHNg85eZY%3D&reserved=0) and [Azure Policy safe deployment features](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fgovernance%2Fpolicy%2Fhow-to%2Fpolicy-safe-deployment-practices&data=05%7C01%7Cnehakulkarni%40microsoft.com%7C132aa96f5b9a4108065908dbd718d258%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638340273857154404%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=AmtvQrzUxz2F6xOJ76%2Bl6zciYrz%2B%2BagUz36G4HKYbdI%3D&reserved=0)).

Reviewer: Looks like this is an outlook link - might want to change it to: https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#install-azure-policy-add-on-for-aks:~:text=exception%20YAML.-,Install%20Azure%20Policy%20Add%2Don%20for%20AKS,-Before%20you%20install

> * Starting January 2024, due to Gatekeeper Upstream removing validation for constraint template contents at create/update time, [the Azure Policy Add-On](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fgovernance%2Fpolicy%2Fconcepts%2Fpolicy-for-kubernetes%23install-azure-policy-add-on-for-aks&data=05%7C01%7Cnehakulkarni%40microsoft.com%7C132aa96f5b9a4108065908dbd718d258%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638340273857129567%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=73Ow8x1Ox8bTw5yOP5P4NIk1R%2FYnaDzhExBBuDbz62g%3D&reserved=0) will now no longer support this. The Azure Policy Add-On will report [‘InvalidConstraint/Template’ compliance reason code](https://learn.microsoft.com/en-us/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons) for detected errors after constraint template admission. This change does not impact [other compliance reason codes](https://learn.microsoft.com/en-us/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons). Customers are encouraged to continue to follow best practices when updating Azure Policy for Kubernetes definitions (i.e. [Gator CLI](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fopen-policy-agent.github.io%2Fgatekeeper%2Fwebsite%2Fdocs%2Fgator%2F&data=05%7C01%7Cnehakulkarni%40microsoft.com%7C132aa96f5b9a4108065908dbd718d258%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638340273857147647%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6VrGYNGtikDsPw3odmyR9WatG4Pe1tlsvDgHNg85eZY%3D&reserved=0) and [Azure Policy safe deployment features](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fgovernance%2Fpolicy%2Fhow-to%2Fpolicy-safe-deployment-practices&data=05%7C01%7Cnehakulkarni%40microsoft.com%7C132aa96f5b9a4108065908dbd718d258%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638340273857154404%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=AmtvQrzUxz2F6xOJ76%2Bl6zciYrz%2B%2BagUz36G4HKYbdI%3D&reserved=0)).

Reviewer: ```suggestion
* Starting January 2024, due to Gatekeeper Upstream removing validation for constraint template contents at create/update time, [the Azure Policy Add-On](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#install-azure-policy-add-on-for-aks:~:text=exception%20YAML.-,Install%20Azure%20Policy%20Add%2Don%20for%20AKS,-Before%20you%20install) will now no longer support this. The Azure Policy Add-On will report [‘InvalidConstraint/Template’ compliance reason code](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons) for detected errors after constraint template admission. This change does not impact [other compliance reason codes](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons). Customers are encouraged to continue to follow best practices when updating Azure Policy for Kubernetes definitions (i.e. [Gator CLI](https://open-policy-agent.github.io/gatekeeper/website/docs/gator/).
```

### Release notes 

> * Bug Fixes 

Reviewer: Each of the bug fixes should have a link to somewhere customers can see more information about that item

  * Fix for when an update with the identity property is issued on an existing AKS Fleet with MSI.
  * Fix for if the system nodepool is set to be dedicated, Vertical Pod Autoscaler (VPA) pods can still be scheduled to system nodes.
  * Fix for if VPA webhook's CABundle is different from the cert stored in secret, the VPA webhook image version will be bumped.
  * Fix where pod age is now calculated from cluster creation time rather than cluster stop time for managed cluster start/stop.
  * In AKS with Azure CNI, there is possible deadlock senario where Continer Network Service (CNS) API is not available. [Aync Delete](https://github.com/Azure/azure-container-networking/tree/master/docs/feature/async-delete) fixes this deadlock issue. The CNI calls to CNS to release an IP address from a Pod asynchronously with a failsafe in such a way that if CNS is unavailable, it can recover these events when it does eventually start.

>   * Fix for NPM Windows crashing on k8s 1.28 with Containerd 1.7.

Reviewer: https://github.com/Azure/AKS/issues/3975

  * Fix for Fleet's override case within AKS for NRGLockdown RestrictionLevel. Previously set to `Unspecified` on create, now updated to reflect the correct level.
* Behavioral Change
  * All AKS managed namespaces now have a "kubernetes.azure.com/managedby:" AKS label 
  * VPA updater and recommender resource requests and limits can now be configured by customer to avoid OOMkill or resource wastage.

>   * [Windows Disable Outbound NAT (Preview)](https://learn.microsoft.com/en-us/azure/aks/nat-gateway#disable-outboundnat-for-windows-preview:~:text=identity%20%24IDENTITY_ID-,Disable%20OutboundNAT%20for%20Windows%20(preview),-Windows%20OutboundNAT%20can) now supports WS2019 and WS2022

Reviewer: ```suggestion
  * [Windows Disable Outbound NAT (Preview)](https://learn.microsoft.com/azure/aks/nat-gateway#disable-outboundnat-for-windows-preview:~:text=identity%20%24IDENTITY_ID-,Disable%20OutboundNAT%20for%20Windows%20(preview),-Windows%20OutboundNAT%20can) now supports WS2019 and WS2022
```

* Component Updates

>   * Bumped agentbaker version to [v0.20231019.2](https://github.com/Azure/AgentBaker/tree/master/vhdbuilder/release-notes/AKSUbuntu/gen2/2204containerd)

Reviewer: Not needed - this is the same as the vhd release for ubuntu
```suggestion
```

>   * AKS Mariner image has been updated to [AKSMariner-202310.19.0](vhd-notes/AKSMariner/202310.19.0.txt).

Reviewer: Mariner images not to be included - only AzureLinux needed 
```suggestion
```

* Starting Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
* AKS sent out an advisory regarding [CVE-2023-29332](https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-29332) on September 13, 2023, which impacts AKS agent nodes. Recommended mitigation is to upgrade AKS cluster and AKS node image. If impacted clusters are not upgraded, AKS will apply mitigation on customer's next cluster update operation including node OS updates and node rolling upgrades, which may cause workload disruption. 

### Release notes 
* Preview Features

>   *[Windows Disable Outbound NAT (Preview)](https://learn.microsoft.com/azure/aks/nat-gateway#disable-outboundnat-for-windows-preview:~:text=identity%20%24IDENTITY_ID-,Disable%20OutboundNAT%20for%20Windows%20(preview),-Windows%20OutboundNAT%20can) now supports WS2019 and WS2022.

Reviewer 2: ```suggestion
  * [Windows Disable Outbound NAT (Preview)](https://learn.microsoft.com/azure/aks/nat-gateway#disable-outboundnat-for-windows-preview:~:text=identity%20%24IDENTITY_ID-,Disable%20OutboundNAT%20for%20Windows%20(preview),-Windows%20OutboundNAT%20can) now supports WS2019 and WS2022.
```

* [Windows containerd v1.7](https://github.com/Azure/AKS/issues/3975) will be the default container runtime for k8s v1.28+ on AKS Windows nodes. Windows Host Process (HPC) containers is GA in Windows containerd v1.7 and it has some [breaking changes](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support#container-mounts).

### Release notes 

* Bug Fixes 

>   * Fix for when an update with the identity property is issued on an existing AKS Fleet with MSI.

Reviewer 3: Needs a bit more clarity, what was the user facing behavior that was visible.
AKS fleet can be ambiguous, refering to the overall set of AKS clusters or Azure Fleet manager?

Author: Removed, confirmed don't need to include

>   * Fix for if the system nodepool is set to be dedicated, Vertical Pod Autoscaler (VPA) pods can still be scheduled to system nodes.

Reviewer 3: Language is a bit off. Better to tweak to

Corrected issue where on tainted/dedicated system pools the Vertical Pod Autoscaler (VPA) deployment could end up on non-system pools.

>   * Fix for if VPA webhook's CABundle is different from the cert stored in secret, the VPA webhook image version will be bumped.

Reviewer 3: Same, as above, needs some language syntax fixes

Fixed issue where a Certificate Authority bundle mismatch could produce an update on the image version of the VPA webhook

  * Fix where pod age is now calculated from cluster creation time rather than cluster stop time for [managed cluster start/stop](https://learn.microsoft.com/en-us/azure/aks/start-stop-cluster?tabs=azure-cli).

>   * In AKS with Azure CNI, there is possible deadlock scenario where Container Network Service (CNS) API is not available. [Aync Delete](https://github.com/Azure/azure-container-networking/tree/master/docs/feature/async-delete) fixes this deadlock issue. The CNI calls to CNS to release an IP address from a Pod asynchronously with a failsafe in such a way that if CNS is unavailable, it can recover these events when it does eventually start.

Reviewer 3: Too much detail, and still not clear on the effect. Would the node network not initialize? A new pod not get an IP? 

Once you confirm you say that it caused by a deadlock between CNS and CNI and that was now fixed, or similar.

Author: new pod not getting ip and deleted pod not releasing ip

>   * Fix for Windows NPM crashes in k8s 1.28 with Containerd 1.7. Bug was a result Windows NPM DaemonSet referencing a file that did not exist in its current directory.[Containerd 1.7](https://github.com/Azure/AKS/issues/3975).

Reviewer 3: Ensure that the GH issue has the resolution/fix-released tag

>   * Fixed fleet clusters, so they will not be correctly set to NRG-Lockdown RestrictionLevel Restricted, instead of Unspecified. Additionally, fleet clusters within one of the undesired Unspecified states will be fixed on reconcile.

Reviewer 3: First part confused me. Are they now being set to restricted or unspecified (if the latter why is that correct)

Author: Typo, should be "now being set to restricted"

* Behavioral Change

>   * All AKS managed namespaces now have a "kubernetes.azure.com/managedby:" AKS label. 

Reviewer 3: This closes and existing GH issue as well

Author: Which issue are you referring to? I don't see one for this?

Reviewer 3: https://github.com/Azure/AKS/issues/1417

>   * VPA updater and recommender resource requests and limits can now be configured by customer to avoid OOMkill or resource wastage.

Reviewer 3: Docs?

Author: removed as pushing to next week

>   * Fix to prevent Open Service Mesh control plane from fighting with AKS admission enforcer.

Reviewer 3: Needs language fix.

Corrected conflict between Open Service Mesh and AKS Admission Enforcer

 * [Windows Disable Outbound NAT (Preview)](https://learn.microsoft.com/azure/aks/nat-gateway#disable-outboundnat-for-windows-preview:~:text=identity%20%24IDENTITY_ID-,Disable%20OutboundNAT%20for%20Windows%20(preview),-Windows%20OutboundNAT%20can) now supports WS2019 and WS2022.

>   * Starting from Kubernetes version 1.29.0, Azure Linux AKS clusters deployed will be cgroupV2 by default, and existing Azure Linux AKS clusters will change from cgroupV1 to cgroupV2 when upgraded to Kubernetes version 1.29.0+.

Reviewer 3: This is an announcement (not something that is changing implicitly and 1.29 doesn't exist yet)

  * Microsoft Defender for Cloud publisher image has been updated to 1.0.68 (now distroless)
  * Microsoft Defender for Cloud OldFileCleaner image has been updated to 1.4.68
* Component Updates
  * Azure Linux image has been updated to [Azure Linux - 202310.26.0](vhd-notes/AzureLinux/202310.26.0.txt).

>   * AKS Ubuntu 18.04 image has been updated to [AKSUbuntu-1804-202310.26.0](vhd-notes/aks-ubuntu/AKSUbuntu-1804/202310.26.0.txt).

Reviewer 3: Deprecated, can remove

Author: Removed AKS Ubuntu 18.04
